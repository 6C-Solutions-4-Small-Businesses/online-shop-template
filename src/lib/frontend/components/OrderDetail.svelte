<script lang="ts">

    import OrderedItemSummaryCard from '$lib/frontend/components/OrderedItemSummaryCard.svelte'
    import type {OrderDetailsPresentation} from '$lib/frontend/presentations/OrderDetailsPresentation'

    export let order: OrderDetailsPresentation

</script>
<div class="p-4 m-2">
    <div class="w-full">
        <p>Voici les articles que vous avez command√©:</p>
        <div class="w-full py-4 grid gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-4">
            {#each order.items as item}
                <OrderedItemSummaryCard
                        id="{item.id}"
                        name={item.name}
                        image={item.image}
                        selectedQuantity={item.selectedQuantity}
                        price={item.price}
                />
            {/each}
        </div>
    </div>
    <p class="text-center">Cliquez <a class="text-amber-500" data-testid="track-delivery-link" href="{order.delivery.trackingUrl}" target="_blank">ici</a>
        <sup>
            <svg class="inline-block text-amber-500 text" {...$$props} height="1em" viewBox="0 0 24 24" width="1em" xmlns="http://www.w3.org/2000/svg">
                <path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7a5 5 0 0 0-5 5a5 5 0 0 0 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1c0 1.71-1.39 3.1-3.1 3.1h-4V17h4a5 5 0 0 0 5-5a5 5 0 0 0-5-5"
                      fill="currentColor"/>
            </svg>
        </sup>
        pour suivre la livraison de votre commande.
    </p>
</div>
