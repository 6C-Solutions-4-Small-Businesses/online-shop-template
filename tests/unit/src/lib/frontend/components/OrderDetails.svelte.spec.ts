import OrderDetail from '$lib/frontend/components/OrderDetail.svelte'
import {cleanup, render, screen} from '@testing-library/svelte'
import {mock} from 'vitest-mock-extended'
import type {OrderDetailsPresentation} from '$lib/frontend/presentations/OrderDetailsPresentation'
import '@testing-library/jest-dom'
import {afterEach, vi} from 'vitest'
import {getOfferCartItems} from '$tests/unit/src/lib/TestHelpers'
import type {DeliveryPresentation} from "$lib/frontend/presentations/DeliveryPresentation";

describe('Order Details Component', () => {

    const items = getOfferCartItems()
    const orderId = 'the order id'
    const deliveryTrackingUrl = 'https://tracking.url'

    beforeEach(() => {
        render(OrderDetail, {
            props: {
                order: mock<OrderDetailsPresentation>({
                    id: orderId,
                    status: 'InProgress',
                    items,
                    delivery: mock<DeliveryPresentation>({
                        trackingUrl: deliveryTrackingUrl,
                    })
                }),
            },
        })
    })

    describe('statically', () => {

        it('should have match snapshot', () => {
            const {container} = render(OrderDetail, {
                props: {
                    order: mock<OrderDetailsPresentation>({
                        items: [mock({
                            id: 'product-id-1',
                            name: 'product-1',
                            price: 100,
                            image: 'an-image-url',
                            description: 'a-product',
                            quantity: 10,
                            isSoldByQuantities: false,
                            unit: 'Centiliter',
                        }), mock({
                            id: 'product-id-2',
                            name: 'product-2',
                            price: 800,
                            image: 'an-image-url',
                            description: 'a-product',
                            quantity: 10,
                            isSoldByQuantities: false,
                            unit: 'Milliliter',
                        }), mock({
                            id: 'product-id-3',
                            name: 'product-3',
                            price: 1200,
                            image: 'an-image-url',
                            description: 'a-product',
                            quantity: 10,
                            isSoldByQuantities: false,
                            unit: 'Gramme',
                        })],
                    })
                }
            })

            expect(container).toMatchSnapshot()
        })
    })

    describe('initially', () => {

        it('should show the link for tracking the delivery', async () => {

            expect(screen.getByTestId('track-delivery-link')).toHaveAttribute('href', deliveryTrackingUrl)
        })

        test.each([
            ...items,
        ])('should show the item $id in the order', async (cartItem: { id: string }) => {

            expect(await screen.findByTestId(cartItem.id)).toBeInTheDocument()
        })
    })

    afterEach(() => {
        cleanup()
        vi.clearAllMocks()
    })
})
