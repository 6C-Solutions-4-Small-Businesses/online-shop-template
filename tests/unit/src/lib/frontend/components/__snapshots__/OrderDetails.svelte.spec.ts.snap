// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Order Details Component > statically > should have match snapshot 1`] = `
<body>
  <div>
    <div
      class="p-4 m-2"
    >
      <div
        class="w-full"
      >
        <p>
          Voici les articles que vous avez commandé:
        </p>
         
        <div
          class="w-full py-4 grid gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-4"
        >
          <div
            class="h-32 xl:h-36 flex bg-white rounded-md shadow-md s-ReROHsK2djRp"
            data-testid="a very good id"
          >
            <img
              alt="Hair Softener"
              class="w-2/5 p-2 h-full product-image object-contain s-ReROHsK2djRp"
              src="https://5jc9cdsla9yvcq5o.public.blob.vercel-storage.com/010181140082.webp"
            />
             
            <div
              class="w-3/5 flex flex-col justify-between hover:bg-orange-100 pl-2 s-ReROHsK2djRp"
            >
              <div
                class="flex justify-between s-ReROHsK2djRp"
              >
                <div
                  class="font-thin flex justify-center items-center s-ReROHsK2djRp"
                >
                  4 x Hair Softener
                </div>
              </div>
               
              <div
                class="flex justify-between h-10 s-ReROHsK2djRp"
              >
                <div
                  class="w-4/12 font-thin flex items-center s-ReROHsK2djRp"
                >
                  $
                  12.34
                </div>
              </div>
            </div>
          </div>
          <!--&lt;OrderedItemSummaryCard&gt;-->
          <div
            class="h-32 xl:h-36 flex bg-white rounded-md shadow-md s-ReROHsK2djRp"
            data-testid="another very good id"
          >
            <img
              alt="Beard Tinter"
              class="w-2/5 p-2 h-full product-image object-contain s-ReROHsK2djRp"
              src="https://5jc9cdsla9yvcq5o.public.blob.vercel-storage.com/011509049063.webp"
            />
             
            <div
              class="w-3/5 flex flex-col justify-between hover:bg-orange-100 pl-2 s-ReROHsK2djRp"
            >
              <div
                class="flex justify-between s-ReROHsK2djRp"
              >
                <div
                  class="font-thin flex justify-center items-center s-ReROHsK2djRp"
                >
                  2 x Beard Tinter
                </div>
              </div>
               
              <div
                class="flex justify-between h-10 s-ReROHsK2djRp"
              >
                <div
                  class="w-4/12 font-thin flex items-center s-ReROHsK2djRp"
                >
                  $
                  4.59
                </div>
              </div>
            </div>
          </div>
          <!--&lt;OrderedItemSummaryCard&gt;-->
        </div>
      </div>
       
      <p
        class="text-center"
      >
        Cliquez 
        <a
          class="text-amber-500"
          data-testid="track-delivery-link"
          href="https://tracking.url"
          target="_blank"
        >
          ici
        </a>
         
        <sup>
          <svg
            class="inline-block text-amber-500 text"
            height="1em"
            order="undefined"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7a5 5 0 0 0-5 5a5 5 0 0 0 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1c0 1.71-1.39 3.1-3.1 3.1h-4V17h4a5 5 0 0 0 5-5a5 5 0 0 0-5-5"
              fill="currentColor"
            />
          </svg>
        </sup>
        
        pour suivre la livraison de votre commande.
      </p>
    </div>
    <!--&lt;OrderDetail&gt;-->
  </div>
  <div>
    <div
      class="p-4 m-2"
    >
      <div
        class="w-full"
      >
        <p>
          Voici les articles que vous avez commandé:
        </p>
         
        <div
          class="w-full py-4 grid gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-4"
        >
          <div
            class="h-32 xl:h-36 flex bg-white rounded-md shadow-md s-ReROHsK2djRp"
            data-testid="product-id-1"
          >
            <img
              alt="product-1"
              class="w-2/5 p-2 h-full product-image object-contain s-ReROHsK2djRp"
              src="an-image-url"
            />
             
            <div
              class="w-3/5 flex flex-col justify-between hover:bg-orange-100 pl-2 s-ReROHsK2djRp"
            >
              <div
                class="flex justify-between s-ReROHsK2djRp"
              >
                <div
                  class="font-thin flex justify-center items-center s-ReROHsK2djRp"
                >
                  function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) =&gt; a[1] = o).catch((o) =&gt; {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  } x product-1
                </div>
              </div>
               
              <div
                class="flex justify-between h-10 s-ReROHsK2djRp"
              >
                <div
                  class="w-4/12 font-thin flex items-center s-ReROHsK2djRp"
                >
                  $
                  1.00
                </div>
              </div>
            </div>
          </div>
          <!--&lt;OrderedItemSummaryCard&gt;-->
          <div
            class="h-32 xl:h-36 flex bg-white rounded-md shadow-md s-ReROHsK2djRp"
            data-testid="product-id-2"
          >
            <img
              alt="product-2"
              class="w-2/5 p-2 h-full product-image object-contain s-ReROHsK2djRp"
              src="an-image-url"
            />
             
            <div
              class="w-3/5 flex flex-col justify-between hover:bg-orange-100 pl-2 s-ReROHsK2djRp"
            >
              <div
                class="flex justify-between s-ReROHsK2djRp"
              >
                <div
                  class="font-thin flex justify-center items-center s-ReROHsK2djRp"
                >
                  function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) =&gt; a[1] = o).catch((o) =&gt; {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  } x product-2
                </div>
              </div>
               
              <div
                class="flex justify-between h-10 s-ReROHsK2djRp"
              >
                <div
                  class="w-4/12 font-thin flex items-center s-ReROHsK2djRp"
                >
                  $
                  8.00
                </div>
              </div>
            </div>
          </div>
          <!--&lt;OrderedItemSummaryCard&gt;-->
          <div
            class="h-32 xl:h-36 flex bg-white rounded-md shadow-md s-ReROHsK2djRp"
            data-testid="product-id-3"
          >
            <img
              alt="product-3"
              class="w-2/5 p-2 h-full product-image object-contain s-ReROHsK2djRp"
              src="an-image-url"
            />
             
            <div
              class="w-3/5 flex flex-col justify-between hover:bg-orange-100 pl-2 s-ReROHsK2djRp"
            >
              <div
                class="flex justify-between s-ReROHsK2djRp"
              >
                <div
                  class="font-thin flex justify-center items-center s-ReROHsK2djRp"
                >
                  function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) =&gt; a[1] = o).catch((o) =&gt; {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  } x product-3
                </div>
              </div>
               
              <div
                class="flex justify-between h-10 s-ReROHsK2djRp"
              >
                <div
                  class="w-4/12 font-thin flex items-center s-ReROHsK2djRp"
                >
                  $
                  12.00
                </div>
              </div>
            </div>
          </div>
          <!--&lt;OrderedItemSummaryCard&gt;-->
        </div>
      </div>
       
      <p
        class="text-center"
      >
        Cliquez 
        <a
          class="text-amber-500"
          data-testid="track-delivery-link"
          target="_blank"
        >
          ici
        </a>
         
        <sup>
          <svg
            class="inline-block text-amber-500 text"
            height="1em"
            order="undefined"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7a5 5 0 0 0-5 5a5 5 0 0 0 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1c0 1.71-1.39 3.1-3.1 3.1h-4V17h4a5 5 0 0 0 5-5a5 5 0 0 0-5-5"
              fill="currentColor"
            />
          </svg>
        </sup>
        
        pour suivre la livraison de votre commande.
      </p>
    </div>
    <!--&lt;OrderDetail&gt;-->
  </div>
</body>
`;
